app/scripts/views/sign_up.js

```diff
@@ -157,15 +157,17 @@ define(function (require, exports, module) {
         chooseWhatToSyncCheckbox: this.broker.hasCapability('chooseWhatToSyncCheckbox'),
         email: prefillEmail,
         error: this.error,
+        isAmoMigration: this.isAmoMigration(),
         isCustomizeSyncChecked: relier.isCustomizeSyncChecked(),
         isEmailOptInVisible: this._isEmailOptInEnabled(),
-        isMigration: this.isMigration(),
         isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled(),
         isSync: isSync,
+        isSyncMigration: this.isSyncMigration(),
         password: prefillPassword,
         serviceName: relier.get('serviceName'),
         shouldFocusEmail: autofocusEl === 'email',
-        shouldFocusPassword: autofocusEl === 'password'
+        shouldFocusPassword: autofocusEl === 'password',
+        signinUri: '/signin'
       };
 
```

Comment:
We'll need some logic to clear the email address whenever the user clicks the signin link from the amo migration text. I think this patch should do it:

```diff
diff --git a/app/scripts/templates/sign_up.mustache b/app/scripts/templates/sign_up.mustache
index 718f4af..7f1be71 100644
--- a/app/scripts/templates/sign_up.mustache
+++ b/app/scripts/templates/sign_up.mustache
@@ -24,7 +24,7 @@
     {{/isSyncMigration}}
 
     {{#isAmoMigration}}
-        <div class="info nudge pad">{{#t}}Have an account with a different email? <a href="%(signinUri)s">Sign in</a>{{/t}}</div>
+        <div class="info nudge pad" id="amo-migration">{{#t}}Have an account with a different email? <a href="%(signinUri)s">Sign in</a>{{/t}}</div>
     {{/isAmoMigration}}
 
     <form novalidate>
diff --git a/app/scripts/views/sign_up.js b/app/scripts/views/sign_up.js
index 3025768..23cf2b1 100644
--- a/app/scripts/views/sign_up.js
+++ b/app/scripts/views/sign_up.js
@@ -128,7 +128,8 @@ define(function (require, exports, module) {
 
     events: {
       'blur input.email': 'onEmailBlur',
-      'blur input.password': 'onPasswordBlur'
+      'blur input.password': 'onPasswordBlur',
+      'click #amo-migration a': 'onAmoSignIn'
     },
 
     getPrefillEmail: function () {
@@ -167,7 +168,7 @@ define(function (require, exports, module) {
         serviceName: relier.get('serviceName'),
         shouldFocusEmail: autofocusEl === 'email',
         shouldFocusPassword: autofocusEl === 'password',
-        signinUri: '/signin'
+        signinUri: this.broker.transformLink('/signin')
       };
 
       if (isSync && this.isInExperiment('syncCheckbox')) {
@@ -241,6 +242,13 @@ define(function (require, exports, module) {
       }
     },
 
+    onAmoSignIn: function () {
+      // The user has chosen to sign in with a different email, clear the
+      // email from the relier so it's not used again on the signin page.
+      this.relier.unset('email');
+      this.$('input[type=email]').val('');
+    },
+
     _isEmailSameAsBouncedEmail: function () {
       var bouncedEmail = this.model.get('bouncedEmail');
 
```